pallets/parachain-staking/src/migrations.rs

```diff
@@ -32,24 +35,80 @@ pub enum StakingStorageVersion {
 	V6,     // Fix delegator replacement bug
 }
 
-#[cfg(feature = "try-runtime")]
-impl StakingStorageVersion {
-	/// The latest storage version.
-	#[allow(dead_code)]
-	fn latest() -> Self {
-		Self::V6
+// Migrate to new StorageVersion paradigm and CandidatePool to be
+// a CountedStorageMap
+pub mod v7 {
+	use super::*;
+	use crate::{CandidatePool, Config, Pallet};
+
+	use frame_support::{
+		generate_storage_alias,
+		pallet_prelude::Weight,
+		storage::migration::{have_storage_value, remove_storage_prefix},
+		traits::{Get, GetStorageVersion, StorageVersion as NewStorageVersion},
+	};
+	use log::info;
+	use sp_runtime::traits::Zero;
+
+	// Get storage items into scope which are removed during this migration
+	generate_storage_alias!(ParachainStaking, CandidateCount => Value<u32>);
+	generate_storage_alias!(ParachainStaking, StorageVersion => Value<StakingStorageVersion>);
+
+	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
```

Comment:
should be possible to have `OnRuntimeUpgrade` implemented here?

Reply:
Yup! [73c4cf4](https://github.com/KILTprotocol/mashnet-node/pull/318/commits/73c4cf4c0de0bbe5d765c81acf038e44e60ba2e5)